Model/Plugins/ShippingAddressManagement.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/**
+ * PAYONE Magento 2 Connector is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PAYONE Magento 2 Connector is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with PAYONE Magento 2 Connector. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * PHP version 5
+ *
+ * @category  Payone
+ * @package   Payone_Magento2_Plugin
+ * @author    FATCHIP GmbH <support@fatchip.de>
+ * @copyright 2003 - 2016 Payone GmbH
+ * @license   <http://www.gnu.org/licenses/> GNU Lesser General Public License
+ * @link      http://www.payone.de
+ */
+
+namespace Payone\Core\Model\Plugins;
+
+use Magento\Quote\Api\Data\AddressInterface;
+use Magento\Quote\Model\ShippingAddressManagement as ShippingAddressManagementOrig;
+use Magento\Framework\Exception\LocalizedException;
+
+/**
+ * Plugin for Magentos ShippingAddressManagement class
+ */
+class ShippingAddressManagement
+{
+    /**
+     * Checkout session
+     *
+     * @var \Magento\Checkout\Model\Session
+     */
+    protected $checkoutSession;
+
+    /**
+     * Quote repository.
+     *
+     * @var \Magento\Quote\Api\CartRepositoryInterface
+     */
+    protected $quoteRepository;
+
+    /**
+     * PAYONE addresscheck request model
+     *
+     * @var \Payone\Core\Model\Risk\Addresscheck
+     */
+    protected $addresscheck;
+
+    /**
+     * Constructor
+     *
+     * @param \Magento\Quote\Api\CartRepositoryInterface $quoteRepository
+     * @param \Magento\Checkout\Model\Session            $checkoutSession
+     * @param \Payone\Core\Model\Risk\Addresscheck       $addresscheck
+     */
+    public function __construct(
+        \Magento\Quote\Api\CartRepositoryInterface $quoteRepository,
+        \Magento\Checkout\Model\Session $checkoutSession,
+        \Payone\Core\Model\Risk\Addresscheck $addresscheck
+    ) {
+        $this->quoteRepository = $quoteRepository;
+        $this->checkoutSession = $checkoutSession;
+        $this->addresscheck = $addresscheck;
+    }
+
+    /**
+     * Execute addresscheck and return the response
+     *
+     * @param  AddressInterface $oAddress
+     * @return array
+     */
+    protected function handleAddresscheck(AddressInterface $oAddress)
```

Comment:
This is somewhat duplicated for Shipping and Billing addresses. Can we avoid the duplication?